common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -111,8 +111,30 @@ public int numConnectionsPerPeer() {
   /** Requested maximum length of the queue of incoming connections. Default is 64. */
   public int backLog() { return conf.getInt(SPARK_NETWORK_IO_BACKLOG_KEY, 64); }
 
-  /** Number of threads used in the server thread pool. Default to 0, which is 2x#cores. */
-  public int serverThreads() { return conf.getInt(SPARK_NETWORK_IO_SERVERTHREADS_KEY, 0); }
+  /**
+   * The configured ratio between number of server threads and number of chunk fetch handler
+   * threads. Default to 1, which sets the size of both thread pools to be equal. Number of
+   * server threads needs to be a multiple of this ratio. See SPARK-29206.
+   */
+  private int getChunkFetchHandlerThreadsRatio() {
+    return conf.getInt("spark.shuffle.server.chunkFetchHandlerThreadsRatio", 1);
```

Comment:
This removes an existing setting and introduces a new one (which would need to use the new config system anyway). Why not simply round up the user's configured number to the next multiple? That's kind of what happens here anyway. Why do we need them to specify this differently?